release_creation/github_client/releases.py

```diff
@@ -0,0 +1,87 @@
+import os
+import requests
+from typing import Dict, List, Optional, Set, Tuple
+from semantic_version import Version, match
+from github import Github
+from github.GitRelease import GitRelease
+from github.GitReleaseAsset import GitReleaseAsset
+
+_GH_SNAPSHOT_REPO = "dbt-labs/dbt-core-snapshots"
+_GH_ACCESS_TOKEN = os.environ["GH_ACCESS_TOKEN"]
+_SNAP_REQ_NAME = "snapshot_requirements"
+
+
+def get_github_client() -> Github:
+    return Github(_GH_ACCESS_TOKEN)
+
+
+def get_latest_release(input_version:str) -> Tuple[Version, Optional[GitRelease]]:
```

Comment:
Naming suggestion for this method - `get_latest_snapshot_release`, given it's tied to `_GH_SNAPSHOT_REPO`.